src/App.tsx

```diff
@@ -1,26 +1,179 @@
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorType.LoadTodo);
+      });
+  }, []);
+
+  const handleFilterChange = (newFilter: FilterType) => {
+    setFilter(newFilter);
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    setLoadingTodoIds(prev => [...prev, todoId]);
+    try {
+      await deleteTodo(todoId);
+      setTodos(prev => prev.filter(todo => todo.id !== todoId));
+    } catch (error) {
+      setErrorMessage(ErrorType.DeleteTodo);
+      inputNameRef.current?.focus();
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(id => id !== todoId));
+    }
+  };
+
+  const handleAddTodo = async (todoTitle: string) => {
+    setTempTodo({
+      id: 0,
+      title: todoTitle,
+      completed: false,
+      userId: USER_ID,
+    });
+    try {
+      const newTodo = await createTodo({ title: todoTitle, completed: false });
+
+      setTodos(prev => [...prev, newTodo]);
+    } catch (error) {
+      setErrorMessage(ErrorType.AddTodo);
+      inputNameRef.current?.focus();
+      throw error;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const handleUpdateTodo = async (todoToUpdate: Todo) => {
+    setLoadingTodoIds(prev => [...prev, todoToUpdate.id]);
+    try {
+      const updatedTodo = await updateTodo(todoToUpdate);
+
+      setTodos(prev =>
+        prev.map(todo => (todo.id === updatedTodo.id ? updatedTodo : todo)),
+      );
+    } catch (err) {
+      setErrorMessage(ErrorType.UpdateTodo);
+      inputNameRef.current?.focus();
+      throw err;
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(id => id !== todoToUpdate.id));
+    }
+  };
+
+  const handletoggleAll = async () => {
+    if (notCompletedTodosCount > 0) {
+      const activeTodos = todos.filter(todo => !todo.completed);
+
+      activeTodos.forEach(todo =>
+        handleUpdateTodo({ ...todo, completed: true }),
+      );
+    } else {
+      todos.forEach(todo => handleUpdateTodo({ ...todo, completed: false }));
+    }
+  };
+
+  const handleClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(todo => {
+      handleDeleteTodo(todo.id);
+    });
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          handleAddTodo={handleAddTodo}
+          setErrorMessage={setErrorMessage}
+          handletoggleAll={handletoggleAll}
```

Comment:
1) Check [naming convention](https://dev.to/kristiyan_velkov/react-js-naming-convention-lcg)
2) syntax error
```suggestion
          onAddTodo={handleAddTodo}
          setErrorMessage={setErrorMessage}
          onToggleAll={handleToggleAll}
```